### PR TITLE
Origin/add be submission

### PR DESCRIPTION
**Pull request by kqpham**
### What this PR does (required):
- I added in submission controllers that handles creating a submission,
- added submission handler for getAllSubmissions this filters so contest owners can only see all submissions relating to the specific contest or showing all submissions relating to the user
- added in getSubmissionId making it so it can only be viewed by either the submission creator or the contest owner
- adjusted a styling within authHeader

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action
-Noticed a error and fixed it as seen in the second video
https://user-images.githubusercontent.com/17209997/121230109-d3bf2d00-c85c-11eb-8153-8c9182d31a3b.mov

https://user-images.githubusercontent.com/17209997/121230164-e6396680-c85c-11eb-88d7-d1b7cbdb75ae.mov

### Any information needed to test this feature (required):
- Posted data to check if it had successfully been applied to the databases
- tested to retrieve data from the database with filters

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/submission.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        } else {
+            res.status(500);
+            throw new Error(
+                "Submission could not be processed at this time, Please Try Again"
+            );
+        }
+    } else {
+        res.status(500);
+        throw new Error(
+            "Contest not found to allow submission"
```
Comment (bonnieli) [inline server/controllers/submission.js:75]: should the error message be that the Contest does not exist?

File: server/controllers/submission.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        //Creating a finding and updating contest to apply submission id to array within contest
+        if (createdSubmission) {
+            let submissionArr = contest.submissions;
+            submissionArr.push(createdSubmission._id);
+            contest.submissions = submissionArr;
+
+            const title =contest.title;
+            const price =contest.price;
+            const description = contest.description;
+            const end_date = contest.end_date
```
Comment (bonnieli) [inline server/controllers/submission.js:39]: i would try to stick to camelCase for variables

```suggestion
            const endDate = contest.end_date
```

File: server/controllers/submission.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+     */
+    else{
+        let contest = await Contest.findById(contest_id);
+        
+        if (!contest) {
+            res.status(404);
+            throw new Error("No contest found for given id");
+        }
+
+        if(contest.creator.toString() === Creator.toString()){
```
Comment (bonnieli) [inline server/controllers/submission.js:111]: are we comparing User objects here? i would suggest comparing IDs instead

File: server/controllers/submission.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            let submissions = await Submission.find({contest_id: contest_id});
```
Comment (bonnieli) [inline server/controllers/submission.js:112]:
```suggestion
            const submissions = await Submission.find({contest_id: contest_id});
```

File: server/controllers/submission.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            res.status(200).json({ submissions: submissions });
+        }
+        else{
+            let submissions = await Submission.find({creator: Creator});
```
Comment (bonnieli) [inline server/controllers/submission.js:116]: same here

File: server/controllers/submission.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  
+    let submission = await Submission.findById(submissionId);
+
+    if (!submission) {
+      res.status(404);
+      throw new Error("No submission found for given id");
+    }
+    else{
+        let contest = await Contest.findById(submission.contest_id);
+        if(creator.toString() === contest.creator.toString() ){
```
Comment (bonnieli) [inline server/controllers/submission.js:134]: same here on comparing objects to strings

**Review by bonnieli — COMMENTED** [review]
Code looks a little messy with too many commented out code and console logs at the moment . Please clean up the files before another code review

File: server/controllers/submission.js (hunk 3 above)
Comment (kqpham) [reply server/controllers/submission.js:111]: That point is comparing IDs

File: server/controllers/submission.js (hunk 4 above)
Comment (kqpham) [reply server/controllers/submission.js:112]: I will fix this right away, thank you!

File: server/controllers/submission.js (hunk 2 above)
Comment (kqpham) [reply server/controllers/submission.js]: I agree, I will fix that right away

File: server/controllers/submission.js (hunk 1 above)
Comment (kqpham) [reply server/controllers/submission.js]: I agree the error message should be contest does not exist

File: server/routes/uploadImg.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ const uploadMultiImage = require("../utils/uploadFiles");
 
 router.post("/images-upload", (req, res) => {
   uploadMultiImage(req, res, (error) => {
-    console.log("uploaded files", req.files);
+    //console.log("uploaded files", req.files);
```
Comment (bonnieli) [inline server/routes/uploadImg.js:10]: remove

File: server/controllers/submission.js (hunk 3 above)
Comment (bonnieli) [reply server/controllers/submission.js:98]: oh i see so `.creator` attribute is an ID? okay not a huge issue but best practice is to rename this to `creatorId` cause it seems like you're comparing objects

**Review by bonnieli — APPROVED** [review]

File: server/controllers/submission.js (hunk 3 above)
Comment (kqpham) [reply server/controllers/submission.js:98]: The .creator attribute was named within the model by someone else! Sorry for the confusion !

File: server/routes/uploadImg.js (hunk 7 above)
Comment (kqpham) [reply server/routes/uploadImg.js]: I have completed this, thank you !

**Pull request merged**